.github/workflows/release.yml

```
@@ -0,0 +1,68 @@
+name: New release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  generate_changelog:
+    runs-on: ubuntu-latest
+    name: create release draft
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: 'Get Previous tag'
+        id: previoustag
+        uses: "WyriHaximus/github-action-get-previous-tag@master"
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: calculate next version
+        id: version
+        uses: patrickjahns/version-drafter-action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate changelog
+        uses: charmixer/auto-changelog-action@8095796
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          future_release: ${{ steps.version.outputs.next-version }}
+
+      - name: Generate changelog for the release
+        uses: charmixer/auto-changelog-action@8095796
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          since_tag: ${{ steps.previoustag.outputs.tag }}
+          future_release: ${{ steps.version.outputs.next-version }}
+          output: CHANGELOGRELEASE.md
+
+      - name: push changelog
+        uses: github-actions-x/commit@v2.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          push-branch: 'master'
+          commit-message: 'update changelog'
+          force-add: 'true'
+          files: CHANGELOG.md
+          name: T-Systems MMS
+          email: frage@t-systems-mms.com
```

Maybe swap these two tasks, looks more natural to me and would correlate corresponding actions

```suggestion
      - name: push changelog
        uses: github-actions-x/commit@v2.6
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          push-branch: 'master'
          commit-message: 'update changelog'
          force-add: 'true'
          files: CHANGELOG.md
          name: T-Systems MMS
          email: frage@t-systems-mms.com
          
      - name: Generate changelog for the release
        uses: charmixer/auto-changelog-action@8095796
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          since_tag: ${{ steps.previoustag.outputs.tag }}
          future_release: ${{ steps.version.outputs.next-version }}
          output: CHANGELOGRELEASE.md
```